modules/gapi/test/gapi_fluid_test.cpp

```diff
@@ -244,6 +244,45 @@ TEST(Fluid, Sum_2_Mats_And_Scalar)
     EXPECT_EQ(0, cv::countNonZero(out_mat != ref_mat));
 }
 
+TEST(Fluid, EqualizeHist)
+{
+    cv::GMat mat;
+    cv::GArray<int> arr;
+
+    cv::GComputation c(cv::GIn(mat, arr), cv::GOut(TEqualizeHist::on(mat, arr)));
+    cv::Mat in_mat(320, 480, CV_8UC1),
+            out_mat(320, 480, CV_8UC1),
+            ref_mat(320, 480, CV_8UC1);
+
+    cv::randu(in_mat, 200, 240);
+    std::vector<int> in_vec(256);
+
+    // Calculate normalized accumulated integral transformation array for gapi
+    for(int i = 0; i < in_mat.rows; ++i)
+        for(int j = 0; j < in_mat.cols; ++j)
+            ++in_vec[in_mat.at<uint8_t>(i, j)];
+
+    for(unsigned int i = 1; i < in_vec.size(); ++i)
+        in_vec[i]+=in_vec[i-1];
+
+    int min = 320*480;
+    for(unsigned int i = 0; i < in_vec.size(); ++i)
+        if(in_vec[i] != 0 && in_vec[i] < min)
+            min = in_vec[i];
+
+    for(auto & el : in_vec)
+        el = cvRound(((float)(el - min) / (float)(320*480 - min))*255);
```

Comment:
Just comment what you do here (building a probability density function)

Reply:
Added, thanks